Add cause to thrown exception in SelfSignedCert
